internal/fleetdb/fleetdb.go

```diff
@@ -57,20 +57,29 @@ func (s *fleetDBImpl) AddServer(ctx context.Context, serverID uuid.UUID, facilit
 			server := sservice.Server{UUID: serverID, Name: serverID.String(), FacilityCode: facilityCode}
 			_, err := s.client.Delete(ctx, server)
 			if err != nil {
+				s.logger.WithFields(logrus.Fields{
+					"error": err,
+				}).Warningf("server enroll failed to rollback server")
```

Comment:
nit: It might be cleaner to simply use `WithError()`:
```
s.logger.WithError(err).Warning("server enroll failed to rollback server")
```

Reply:
+1 Plus you'll be more consistent with our usage of `WithError` elsewhere in the codebase.